src/counts.jl

```diff
@@ -388,6 +388,9 @@ of occurrences.
 """
 countmap(x::AbstractArray{T}; alg = :auto) where {T} = addcounts!(Dict{T,Int}(), x; alg = alg)
 countmap(x::AbstractArray{T}, wv::AbstractVector{W}) where {T,W<:Real} = addcounts!(Dict{T,W}(), x, wv)
+countmap(x) = addcounts!(Dict{eltype(x),Int}(), x; alg = :dict)
```

Comment:
Should the default be auto or dict?

Reply:
Ah right it should be `:auto`. I guess @Moelf used `:dict` because the PR doesn't currently extend `:dict` methods to general iterables. Is there a reason why this wouldn't be possible?

Reply:
the auto will land in this line where `length` is needed:
https://github.com/JuliaStats/StatsBase.jl/blob/983bf021be7521c399dfe4d25d174667344bc8fb/src/counts.jl#L358

Reply:
This method is only used when weights are provided AFAICT (and I said I don't think we should allow passing two non-vector iterators in that case).

Reply:
That said, `addcounts_radixsort` calls `sort`, which only works on vectors. But you could do `sort!(collect(x))` for other iterators.

Reply:
@nalimilan makes sense. I already removed the two non-vector methods. Now switched back to `:auto` by default.